### DISCUSSION TITLE
Gloss MAC issue for live coding: can this be resolved?

### DISCUSSION THREAD

**Discussion by reubenharry** [ad-si/Brillo#13] [General]
https://github.com/turion/essence-of-live-coding-tutorial/issues/3 discusses an issue I've also had with gloss crashing on mac when run in the way you need to run it for the essence-of-live-code project to work. I'm curious whether Brillo will offer (or already offers) a path to resolving this

**Comment by ad-si** [comment]
It might be fixed. The errors seem to happen with the GLUT backend and gloss still used GLUT even when you activated GLFW as the backend. Brillo *only* uses GLFW and so might not have those issues. Can you try it out?

**Reply by reubenharry** [reply]
Thanks for your reply!

I tried swapping out gloss for brillo in essence-of-live-coding. The types check, and things don't crash when I load up the repl, but brillo doesn't seem to launch the window.

For example, if I define:

```haskell
main = display
  (InWindow "Nice Window" (200, 200) (10, 10))
  white
  (Circle 80)
```

and then do `cabal repl` and run `main`, nothing happens. If this is resolved, then since everything else is working, I should be able to update essence-of-live-code to work on mac.

**Reply by ad-si** [reply]
Do you have it as branch on GitHub so I can check it out and play with it?

**Reply by reubenharry** [reply]
Just uploaded. See the file `Main.hs`: https://github.com/reubenharry/essence-of-live-coding-tutorial/blob/master/Main.hs. I used ghc 9.6.6.

**Reply by ad-si** [reply]
Mh, with `cabal run` it starts correctly, but with `ghcid` nothing happens (not even an error). Maybe  @turion has some insights?
